components/espressif/esp32-wroom-32.stanza

```diff
@@ -0,0 +1,277 @@
+; When renaming the stanza package name below, also rename it in the folder stanza.proj
+; See docs.jitx.com for help
+#use-added-syntax(jitx)
+defpackage ocdb/espressif/esp32-wroom-32:
+  import core
+  import collections
+  import jitx
+
+  import jitx/commands
+  import ocdb/defaults
+  import ocdb/land-patterns
+  import ocdb/tolerance
+  import ocdb/box-symbol
+  import ocdb/bundles
+  import ocdb/land-protrusions
+  import ocdb/design-vars
+  import ocdb/generic-components
+  import ocdb/property-structs
+
+; ================================================== Component ==================================================
+public defn component () :
+  component(4,true,false) 
+
+public defn component (pcb-antenna?:True|False) :
+  component(4,pcb-antenna?, false) 
+
+public defn component (memory:Int, pcb-antenna?:True|False) :
+  component(memory, pcb-antenna?, false) 
+
+public pcb-component component (memory:Int, pcb-antenna?:True|False, high-temp:True|False) :
+  var mem-text:String
+  switch {memory == _} :
+    4 : mem-text = ", 4MB"
+    8 : mem-text = ", 8MB"
+    16 : mem-text = ", 16MB"
+    else: fatal("Invalid memory size of %_ passed to esp32-wroom-32. Valid sizes are 4, 8, and 16" % [memory])
+
+  var package-string = "UE"
+  var package-desc = ", UMCX"
+  if pcb-antenna? :
+    package-string = "E" 
+    package-desc = ", PCB Antenna"
+
+  var temp-text = "N"
+  if high-temp:
+    if memory == 4 :
+      temp-text = "H"
+      mem-text = append(", High Temperature", mem-text)
+      property(self.rated-temperature) = RatedTemperature(min-max(-40.0, 105.0))
+    else : fatal("Invalid memory size of %_ for high temp version passed to esp32-wroom-32. Valid high temp memory option is 4." % [memory])
+  else:
+    property(self.rated-temperature) = RatedTemperature(min-max(-40.0, 85.0))
+
+  name = string-join(["ESP32-WROOM-32" package-string "-" temp-text memory])
+  mpn = string-join(["ESP32-WROOM-32" package-string "-" temp-text memory])
+  description = string-join(["Module, ESP32, "package-desc mem-text ", WiFi 802.11b/g/n, Bluetooth v4.2 +EDR, Class 1, 2 and 3 Transceiver, Surface Mount"])
+  manufacturer = "Espressif" 
+  reference-prefix = "U"
+
+  pin EN 
+  port IO : pin[[0 2 4 5 12 13 14 15 16 17 18 19 21 22 23 25 26 27 32 33 34 35]]
+  pin RXD0 
+  pin TXD0 
+
+  pin-properties :
```

Comment:
It's been awhile since I've done this - is it valid to mix pin definitions and pin properties like this? (or necessary?)

Reply:
assigning IO explicitly is required, but I removed the others